src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -214,25 +214,32 @@ protected function getAllowedAttributes(string|object $classOrObject, array $con
         }
 
         $groups = $this->getGroups($context);
+        $groupsHasBeenDefined = [] !== $groups;
+
+        $classMetadata = $this->classMetadataFactory->getMetadataFor($classOrObject);
+        $class = $classMetadata->getName();
+
+        $groups = array_merge($groups, ['Default', (false !== $nsSep = strrpos($class, '\\')) ? substr($class, $nsSep + 1) : $class]);
```

Comment:
Consider placing this on line 218
Define the $classMetadata and $class above line 216.